rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -258,6 +258,35 @@ inline void microphone_device::variable_byteswap(const void* src, void* dst)
 	}
 }
 
+inline std::vector<u8> microphone_device::convert_16_bit_pcm_to_float(const std::vector<u8>& buffer, usz num_bytes)
+{
+	ensure(num_bytes % 2 == 0);
+	ensure(num_bytes < buffer.size());
+
+	std::vector<u8> float_buffer;
```

Comment:
For performance it would make more sense to just use a float_buf member (similar to temp_buf) and resize it here with
```
constexpr usz float_buf_size = 2 * inbuf_size;
static_assert((float_buf_size % sizeof(u32)) == 0);

float_buf.resize(float_buf_size);
ensure(num_bytes * 2 <= float_buf.size());

u32* dst = reinterpret_cast<u32*>(float_buf.data());
```
Then return the num_bytes * 2 and use that as bufsize for rbuf_dsp.write_bytes

Reply:
Then the function won't be static anymore

Reply:
well then don't make the function static ?